tests/expectations/metrics/test_core.py

```diff
@@ -156,6 +160,70 @@ def test_column_value_lengths_min_metric_pd():
     assert results == {desired_metric.id: 8}
 
 
+def test_column_quoted_name_type_sa(sa):
+    engine = build_sa_engine(
+        pd.DataFrame(
+            {
+                "names": [
+                    "Ada Lovelace",
+                    "Alan Kay",
+                    "Donald Knuth",
+                    "Edsger Dijkstra",
+                    "Guido van Rossum",
+                    "John McCarthy",
+                    "Marvin Minsky",
+                    "Ray Ozzie",
+                ]
+            }
+        ),
+        sa,
+    )
+
+    metrics: Dict[Tuple[str, str, str], MetricValue] = {}
+
+    table_columns_metric: MetricConfiguration
+    results: Dict[Tuple[str, str, str], MetricValue]
+
+    table_columns_metric, results = get_table_columns_metric(engine=engine)
+    metrics.update(results)
+
+    table_columns_metric: MetricConfiguration = MetricConfiguration(
+        metric_name="table.columns",
+        metric_domain_kwargs=dict(),
+        metric_value_kwargs=None,
+    )
+    table_columns_metric_id: Tuple[str, str, str] = table_columns_metric.id
+
+    column_name = "names"
+
+    metrics[table_columns_metric_id] = [quoted_name(value=column_name, quote=True)]
```

Comment:
Could we add a test case where we try and fail to get a metric from the uppercase column "NAMES" using quoted_name?

Reply:
@talagluck We just need a user-acceptance test on `Snowflake` -- I already reached out to the user to do that.